### DISCUSSION TITLE
Allow whitelisted emails to create an account

### DISCUSSION THREAD

**Discussion by dwesty17** [directus/directus#3956] [Feature Requests]
Basically exactly what @benhaynes mentioned [here](https://github.com/directus/v8-archive/issues/664#issuecomment-445570679). This would be a super helpful enterprise feature for us at Sonder.

The approach is simple. Just create a new user when someone tries to authenticate through an SSO provider and their email address matches a whitelisted domain. The whitelisted domain can be configured through a new environment variable.

I took shot at a [PR](https://github.com/directus/directus/pull/3955) for this. I'd also be happy to open new PRs for the following (or add it to the open PR)...
- Edit the docs to explain this new feature
- Add new form that prompts user to enter basic info like name upon first log in
- Update the error screen to have a more helpful message upon first log in

For the last point, I was just thinking that it'd be helpful to change the error message to say, "Your account don't have any permissions yet. Ask an Admin to assign you a role so you can get started," (or something along those lines) when a user without a role logs in.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
